File: Interfaces/ITrRouteService.cs
using WMS_WEBAPI.DTOs;

namespace WMS_WEBAPI.Interfaces
{
    public interface ITrRouteService
    {
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetAllAsync();
        Task<ApiResponse<TrRouteDto>> GetByIdAsync(long id);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetByLineIdAsync(long lineId);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetByStockCodeAsync(string stockCode);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetBySerialNoAsync(string serialNo);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetBySourceWarehouseAsync(string sourceWarehouse);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetByTargetWarehouseAsync(string targetWarehouse);
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetActiveAsync();
        Task<ApiResponse<IEnumerable<TrRouteDto>>> GetByQuantityRangeAsync(decimal minQuantity, decimal maxQuantity);
        Task<ApiResponse<TrRouteDto>> CreateAsync(CreateTrRouteDto createDto);
        Task<ApiResponse<TrRouteDto>> UpdateAsync(long id, UpdateTrRouteDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
    }
}

File: WebApi/Interfaces/ITrRouteService.ts
import { ApiResponse } from '../Models/ApiResponse';
import { TrRouteDto, CreateTrRouteDto, UpdateTrRouteDto } from '../Models/TrRouteDtos';

export interface ITrRouteService {
  getAll(): Promise<ApiResponse<TrRouteDto[]>>;
  getById(id: number): Promise<ApiResponse<TrRouteDto>>;
  getByLineId(lineId: number): Promise<ApiResponse<TrRouteDto[]>>;
  getByStockCode(stockCode: string): Promise<ApiResponse<TrRouteDto[]>>;
  getBySerialNo(serialNo: string): Promise<ApiResponse<TrRouteDto[]>>;
  getBySourceWarehouse(sourceWarehouse: string): Promise<ApiResponse<TrRouteDto[]>>;
  getByTargetWarehouse(targetWarehouse: string): Promise<ApiResponse<TrRouteDto[]>>;
  getActive(): Promise<ApiResponse<TrRouteDto[]>>;
  getByQuantityRange(minQuantity: number, maxQuantity: number): Promise<ApiResponse<TrRouteDto[]>>;
  create(createDto: CreateTrRouteDto): Promise<ApiResponse<TrRouteDto>>;
  update(id: number, updateDto: UpdateTrRouteDto): Promise<ApiResponse<TrRouteDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
}